pkg/machine/config.go

```diff
@@ -19,6 +20,65 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+type GuestOS uint32
+
+const (
+	CoreOS GuestOS = iota
```

Comment:
CoreOS is a Linux distribution, not sure why it needs a separate "GuestOS" from others (like Fedora, etc)

Reply:
I'm not sure I understand this comment, please can you elaborate?

Reply:
Why are "CoreOS" and "GenericLinux" (and so on, "Fedora" or "CentOS") different guest operating systems ?

But I think you explained it above, it _is_ about the distribution and not only about the operating system.

Reply:
Yup, there are things that make assumptions about, for example, dnf or systemd, that are per OS (which, in the Linux world, typically means distro), that someone may want to disentangle at some point, so I wanted to leave that as an extension point.  Happy to make it simply Linux / FreeBSD [/ Windows / Fuchsia / whatever] if this isn't something that people want.

Reply:
That would be my preference, the rest is just different images IMHO.